public/app/features/dashboard/history/history.ts

```diff
@@ -0,0 +1,247 @@
+///<reference path="../../../headers/common.d.ts" />
+
+import './history_srv';
+
+import _ from 'lodash';
+import angular from 'angular';
+import moment from 'moment';
+
+import {DashboardModel} from '../model';
+import {HistoryListOpts, RevisionsModel} from './models';
+
+export class HistoryListCtrl {
+  appending: boolean;
+  dashboard: DashboardModel;
+  delta: { basic: string; html: string; };
+  diff: string;
+  limit: number;
+  loading: boolean;
+  max: number;
+  mode: string;
+  orderBy: string;
+  revisions: RevisionsModel[];
+  selected: number[];
+  start: number;
+
+  /** @ngInject */
+  constructor(private $scope,
+              private $rootScope,
+              private $window,
+              private $q,
+              private contextSrv,
+              private historySrv) {
+    $scope.ctrl = this;
+
+    this.appending = false;
+    this.diff = 'basic';
+    this.limit = 10;
+    this.loading = false;
+    this.max = 2;
+    this.mode = 'list';
+    this.orderBy = 'version';
+    this.selected = [];
+    this.start = 0;
+
+    this.resetFromSource();
+
+    $scope.$watch('ctrl.mode', newVal => {
+      $window.scrollTo(0, 0);
+      if (newVal === 'list') {
+        this.reset();
+      }
+    });
+
+    $rootScope.onAppEvent('dashboard-saved', this.onDashboardSaved.bind(this));
+  }
+
+  addToLog() {
+    this.start = this.start + this.limit;
+    this.getLog(true);
+  }
+
+  compareRevisionStateChanged(revision: any) {
+    if (revision.checked) {
+      this.selected.push(revision.version);
+    } else {
+      _.remove(this.selected, version => version === revision.version);
+    }
+    this.selected = _.sortBy(this.selected);
+  }
+
+  compareRevisionDisabled(checked: boolean) {
+    return (this.selected.length === this.max && !checked) || this.revisions.length === 1;
+  }
+
+  formatDate(date) {
+    date = moment.isMoment(date) ? date : moment(date);
+    const format = 'YYYY-MM-DD HH:mm:ss';
+
+    return this.dashboard.timezone === 'browser' ?
```

Comment:
Just for future reference, there is a formatDate function on the dashboard model that does this, so checking timezone and having date formats all over the code base is not needed. All date formatting is done via this function:

https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/model.ts#L247